### ISSUE TITLE
Decimal type support

### ISSUE THREAD

**Issue by umitanuki** [alpacahq/marketstore#37]
I want to put decimal column into marketstore to store crypto order books.  Floating point types won't work because of the precision issues.

For the time being we could work around with variable-length record with string type column, but definitely it'll be great if we support decimal type column natively with potentially arithmetic operations.

**Comment by glycerine** [comment]
See https://www.cockroachlabs.com/blog/apd-arbitrary-precision-decimal-package/  and  https://github.com/cockroachdb/apd

**Comment by umitanuki** [comment]
Oh that's a great suggestion

**Comment by ackleymi** [comment]
Also check https://github.com/shopspring/decimal, the most popular decimal package in go. I used it in go-finance, but wish I had some more feedback from the community on how practical/clean the syntax is.

**Comment by umitanuki** [comment]
Yeah actually we are using shopstring/decimal in our main backend system too.  All we need is to add a few conversion routine in the core and define the official name for decimal as the type name so the core system recognizes it.
